### DISCUSSION TITLE
🎁 ✉️ 📆  UI Kitten Release Notes & News

### DISCUSSION THREAD

**Discussion by artyorsh** [akveo/react-native-ui-kitten#1337] [General]
- 📰 This is a read-only UI Kitten newsletter.
- 📫 Only latest news and release notes.
- 🚀 Subscribe (on the right) to stay tuned!

**Comment by artyorsh** [comment]
## 🎉 v4.1.0 (2019-07-11)

**Release Notes:**

Highlights 🚀

✅ 🌐 **React Native Web** support. [**Demo**](https://i.giphy.com/media/egvol9MmqLOxNfZkAK/source.gif)

✅ 📱New **Modal** component with developer-friendly API

✅ **Input** icon press handler
```jsx
const onIconPress = () => {
  // Handle Icon Press
};

<Input onIconPress={this.onIconPress}/>
```

Lots of other [**bug fixes and improvements**](https://github.com/akveo/react-native-ui-kitten/blob/master/CHANGELOG.md#410-2019-07-10)!

**Comment by 32penkin** [comment]
## 🎉 v4.2.0-beta.2 (2019-08-30)

**Release Notes:**

Highlights 🚀

✅ [**Select**](https://akveo.github.io/react-native-ui-kitten/docs/components/select/overview#select) component

✅ [**Drawer**](https://akveo.github.io/react-native-ui-kitten/docs/components/drawer/overview#drawer) component

✅ [**Spinner**](https://akveo.github.io/react-native-ui-kitten/docs/components/spinner/overview#spinner) component

✅ OverflowMenu component is based on Menu now

✅ [**Icon**](https://akveo.github.io/react-native-ui-kitten/docs/components/icon/overview#icon) component

✅ RTL support!

✅ Meet the brand-new [**Eva-Icons Module**](https://akveo.github.io/react-native-ui-kitten/docs/guides/eva-icons#setup-icons-module) with the pre-setup icons packs!

✅ It is also now possible to [**quickly and easily create**](https://akveo.github.io/react-native-ui-kitten/docs/guides/start-a-new-app#starting-a-new-app) projects with react-native-ui-kitten!

✅ The documentation has expanded with even more examples.

Lots of other [**bug fixes and improvements**](https://github.com/akveo/react-native-ui-kitten/blob/master/CHANGELOG.md#420-2019-08-30)!

**Comment by artyorsh** [comment]
## 🎉 v4.2.0 (2019-09-27)

**Release Notes:**

Highlights 🚀

📝 **Global Font Family**: Now it is possible to set `fontFamily` globally as mapping variable
```jsx
import React from 'react';
import { mapping, light as theme } from '@eva-design/eva';
import { ApplicationProvider } from 'react-native-ui-kitten';

const strictTheme = { ['text-font-family']: 'YOUR-FONT-FAMILY' };
const customMapping = { strict: scrictTheme };

// ...

<ApplicationProvider mapping={mapping} theme={theme} customMapping={customMapping} />
```

📆 [**Calendar**](https://akveo.github.io/react-native-ui-kitten/docs/components/calendar) and [**Datepicker**](https://akveo.github.io/react-native-ui-kitten/docs/components/datepicker) components.
Supports ranges, locales, and differend date objects like [moment](https://momentjs.com)

🔥 New component stylings provided by updated [Eva Design System](https://eva.design)

✅ Toggle now can accept `text` property

🐛 Fixed [react-native template modules](https://akveo.github.io/react-native-ui-kitten/docs/guides/start-a-new-app)
```bash
# Simply run the command below to initialize application based on UI Kitten
npx react-native init AwesomeApp --template react-native-template-ui-kitten
```

📚 Documentation examples were refactored to JavaScript

✅ All components from beta release were updated with bug fixes. Thanks for your feedback! 

Lots of other [**bug fixes and improvements**](https://github.com/akveo/react-native-ui-kitten/blob/master/CHANGELOG.md#420-2019-09-27)!

**Comment by artyorsh** [comment]
## 🎉 Documentation for v4.2.0 (2019-10-11)

📚 Documentation was updated with tons of clickable examples

👇 https://akveo.github.io/react-native-ui-kitten/docs/components/components-overview

**Notes:**

Live examples are running with React Native Web and can contain some bugs with web states like hover and focus. This means it works perfect on mobile, but not on the web. This will be fixed soon with #663

**Comment by artyorsh** [comment]
## 🎉 v4.3.0-beta.1 (2019-10-17)

**Release Notes:**

Highlights 🚀

🌐 **React Native Web states**: Now components support hover and focus states when using in the web! Just open [documentation](https://akveo.github.io/react-native-ui-kitten/docs/components/components-overview) to see how it feels 😍 

📆 **Calendar** now supports setting a first day of week. Notice, that this feature is also available for  `RangeCalendar`, `Datepicker` and  `RangeDatepicker` components. [See it in action](https://akveo.github.io/react-native-ui-kitten/docs/components/calendar/overview#calendar)

🐛 Fixed [Calendar navigation](https://github.com/akveo/react-native-ui-kitten/issues/680) when passing `min` property

🐛 Fixed [ButtonGroup render issues](https://github.com/akveo/react-native-ui-kitten/issues/516) 

🐛 Fixed [Select initial value](https://github.com/akveo/react-native-ui-kitten/pull/684) 

Lots of other [**bug fixes and improvements**](https://github.com/akveo/react-native-ui-kitten/blob/master/CHANGELOG.md#430-beta1-2019-10-17)!

#### Mobile backend Bundles 📱

We also glad to present the new products of Akveo team! See ready-to-go solution with [Mobile Backend Bundles](https://bit.ly/2MjefPY)

#### Updade:
```
npm i react-native-ui-kitten@beta @eva-design/eva@beta
```

**Comment by artyorsh** [comment]
## 🎉 v4.3.0 (2019-12-04)

**Release Notes:**

Highlights 🚀

📚 **Simplified Getting Started Guide**: We have completely reworked our [**Getting Started Guide**](https://akveo.github.io/react-native-ui-kitten/docs/guides/getting-started). Now it includes step-by-step articles to build React Native app including [Branding](https://akveo.github.io/react-native-ui-kitten/docs/guides/branding), [Navigation](https://akveo.github.io/react-native-ui-kitten/docs/guides/configure-navigation) and more. 

💅 **Unified Eva 1.3 Design**: Eva 1.3 brings unified, pure design system stylings to each component to make your app even more perfect.

✅ **[Card component](https://akveo.github.io/react-native-ui-kitten/docs/components/card)**: New UI Kitten component.

Lots of other [**bug fixes and improvements**](https://github.com/akveo/react-native-ui-kitten/blob/master/CHANGELOG.md#430-2019-12-04)

<hr>

⚠️ **react-native-ui-kitten package deprecated**

We deprecate `react-native-ui-kitten` package in order to use `@ui-kitten/components`.
In fact, the new package contains same source code as an old. In order to migrate, consider following the steps below:

```
npm un react-native-ui-kitten && npm i @ui-kitten/components
```

```diff
- import { ApplicationProvider, Layout, Button } from 'react-native-ui-kitten';
+ import { ApplicationProvider, Layout, Button } from '@ui-kitten/components';
```
In case you have no `react-native-svg` package installed:
```
npm i react-native-svg && cd ios && pod install
```

We're going to maintain an old package till UI Kitten v5 so you may delay the migration.

**Comment by artyorsh** [comment]
## 🎉 v4.3.2 (2019-12-13)

**Release Notes:**

Highlights 🚀

🐛Resolve web builds. You don't need custom webpack config to run UI Kitten modules anymore. Furthermore, now it works properly in Safari!
(#763, #729, #692)

🐛Fix performance of Modals, Pickers, and Calendar. [Check them out](https://akveo.github.io/react-native-ui-kitten/docs/components/datepicker)!
(#641)

<hr>

**Upgrade:**
```bash
npm i @ui-kitten/components@latest @ui-kitten/eva-icons@latest @eva-design/eva@latest
```

And restart your app with clearing cache.

**Comment by artyorsh** [comment]
## 🎉 v4.4.0-beta.1 (2020-01-03)

**Highlights** 🚀

✅ **Autocomplete**: new UI Kitten component. [Docs](https://akveo.github.io/react-native-ui-kitten/docs/components/autocomplete)
(#696, #750)

💅 **Styling & Theming**: add useTheme and useStyleSheet hooks. Use useTheme to accept a current theme, and useStyleSheet to create styles in a way similar to`StyleSheet.create`.
[See docs](https://akveo.github.io/react-native-ui-kitten/docs/components/themed-component) for more details

```js
import { useStyleSheet } from '@ui-kitten/components';

const MyComponent = () => {
  const styles = StyleSheet.create();

  return (
    <View style={styles.container} />
  );
};

const StyleSheet = useStyleSheet({
  container: { backgroundColor: 'color-primary-default' }, 
});
```

(#676, #752)

<hr>

**Upgrade:**
```bash
npm i @ui-kitten/components@beta @ui-kitten/eva-icons@beta
```

And restart your app with clearing cache.

**Comment by artyorsh** [comment]
## 🎉 v4.4.0-beta.2 (2020-01-16)

**Highlights** 🚀

📆 **Datepicker**: add `label`, `caption` and `captionIcon` properties (#802)
✅ Add show and hide methods to Modal and Popover components (#808, #813)
✅ Add focus, blur and isFocused methods to Input components.

🐛 **Modal**: Content updates on Android (#807)
🐛 **Autocomplete**: Runtime crash on Android (#800)

<hr>

🎉 **Demo App**: [Kitten Tricks](https://github.com/akveo/kittenTricks) was completely rewritten with functional components and React Hooks. It also includes lots of other significant features:
- Now it is runnable on web
- Expo SDK 36
- React Navigation 5
- Latest UI Kitten
- Project structure was upgraded for easier screens reusing

<hr>

**Upgrade:**
```bash
npm i @ui-kitten/components@beta @ui-kitten/eva-icons@beta
```

And restart your app with clearing cache.

**Comment by artyorsh** [comment]
## 🎉 v4.4.0-beta.3 (2020-01-21)

**Highlights** 🚀

📆**Calendar and Datepicker**: Add ability to specify date format in Calendar and Datepicker components
(#815)
```js
import React from 'react';
import { Datepicker, NativeDateService } from '@ui-kitten/components';

const dateService = new NativeDateService('en', { format: 'DD.MM.YYYY' });

export const DatepickerDateFormatShowcase = () => {
  const [selectedDate, setSelectedDate] = React.useState(null);

  return (
    <Datepicker date={selectedDate} onSelect={setSelectedDate} dateService={dateService} />
  );
};
```

🐛 **Select**: Fix props mutation issue when using with `Object.freeze` (#804)
💣 **BREAKING CHANGE**: useStyleSheet hook was reimplemented due to not following Rules of Hooks
(#831)

**New syntax:**
```diff
import React from 'react';
- import { Layout, useStyleSheet } from '@ui-kitten/components';
+ import { Layout, StyleService, useStyleSheet } from '@ui-kitten/components';

export const UseStyleSheetSimpleUsageShowcase = () => {
-  const styles = StyleSheet.create();
+ const styles = useStyleSheet(themedStyles);

  return (
    <Layout style={styles.container} />
  );
};

- const StyleSheet = useStyleSheet({
+ const themedStyles = StyleService.create({
  container: {
    flex: 1,
    backgroundColor: 'color-success-default', // <-- Eva Theme Variable
  },
});
```

**Upgrade:**
```bash
npm i @ui-kitten/components@beta @ui-kitten/eva-icons@beta
```

And restart your app with clearing cache.

**Comment by artyorsh** [comment]
## 🎉 v4.4.0 (2020-02-10)

**Highlights** 🚀

With this release, we bring new Eva Design System package which will make your app look like material. Try it out by switching `@eva-design/eva` imports to `@eva-design/material` with no more changes 🤯

We also introduce a bare-minimal application that already runs Eva Material. [Check it out!](https://github.com/artyorsh/ui-kitten-material) or try running [Kitten Tricks](https://github.com/akveo/kittenTricks) to see how it looks!

[<img src="https://i.imgur.com/X8O4GHb.png">](https://github.com/artyorsh/ui-kitten-material)

🙀**Demo App [Kitten Tricks](https://github.com/akveo/kittenTricks)** is now runnable in both Expo and React Native environments and supports `expo eject` with no need to refactor expo-specific modules.

🐛 **Modal**: state update during render (#847)
🐛 **Input**: extra vertical padding on Android (#609)

[Complete changelog](https://github.com/akveo/react-native-ui-kitten/blob/master/CHANGELOG.md#440-2020-02-10)

**Upgrade:**
```bash
npm i @ui-kitten/components@latest @ui-kitten/eva-icons@latest @eva-design/eva@latest
```

And restart your app with clearing cache.

**Comment by artyorsh** [comment]
## 🎉 v4.4.1 (2020-02-24)

**Highlights** 🚀

With this release, we introduce a package to help you improve performance of application loading by processing Eva during the build time.

#### Integration ([Guide](https://akveo.github.io/react-native-ui-kitten/docs/guides/improving-performance#improving-performance))

1. `yarn add @ui-kitten/components@latest && yarn add -D @ui-kitten/metro-config`
2. Add or modify **metro.config.js** with following configuration
```diff
+ const MetroConfig = require('@ui-kitten/metro-config');

+ const evaConfig = { 
+   evaPackage: '@eva-design/eva', // Required (Can be `@eva-design/material`)
+   customMappingPath: './custom-mapping.json', // Optional
+ };

+ module.exports = MetroConfig.create(evaConfig, {
+   // Whatever was in previous metro.config.js 
+ });
```
3. Modify props of ApplicationProvider
```diff
- import { mapping, light } from '@eva-design/eva';
+ import * as eva from '@eva-design/eva'; 
- import { customMapping } from './custom-mapping.js'

export default = () => (
- <ApplicationProvider mapping={mapping} theme={light} />
+ <ApplicationProvider {...eva} theme={eva.light} />
);
```
4. Restart your app with clearing cache.

**Comment by artyorsh** [comment]
## 🎉 v5.0.0-alpha.1 (2020-04-08)

**Highlights** 🚀

With this release, we introduce a completely rethought component APIs, improving developer experience by structuring components in a declarative way. These changes also open more capabilities in component styling, providing you a flexible APIs.

We also focused on the documentation. Now it contains more live examples with a detailed description on the implementation. [Check it out](https://akveo.github.io/react-native-ui-kitten/docs/components/button/overview#button)!

[**Changelog**](https://github.com/akveo/react-native-ui-kitten/blob/next/CHANGELOG.md#500-alpha1-2020-04-08) | [**Migration**](https://akveo.github.io/react-native-ui-kitten/docs/migration/4x500-migration#4x--500-ui-kitten-migration)

**Comment by artyorsh** [comment]
## 🎉 v5.0.0 (2020-05-12)

**Highlights** 🚀

Bring stability into v5 by closing tons of the open issues and adding small requested features.

💣 **BREAKING**: refactor `styled` function to be decorator ([#024ac83](https://github.com/akveo/react-native-ui-kitten/pull/1055/commits/024ac83d4a0e35d2916e19d00edb065e2764396d) Closes #1004)
Despite this is a breaking change, this helps us to keep the existing code base much cleaner by removing of `styledComponentName` statics. For the end user this also leads to better type definitions when using refs.

🐛 **Bug Fixes**:
- Runtime crashes of several components when using latest react-native-svg versions (#1056 #1042 #1005)
- Modal: proper update of nested components (#826)
- Input: different height comparing to other components on Android (#1006)
- Typescript: Fix missing type definitions. (#1030 #1024)

📚 **Documentation**:
- Improve live examples loading performance. (#1043)

🚀 **Features**:
- Avatar: add `ImageComponent` property. (#999)
- Datepicker: add `autoDismiss` property` (#1066)
- Metro Config: the package now [includes](https://akveo.github.io/react-native-ui-kitten/docs/guides/improving-performance#command-line-interface) a command line interface to compile mappings manually. This, for example, solves the case when it's needed to build an app within the CI environment.

[**Changelog**](https://github.com/akveo/react-native-ui-kitten/blob/master/CHANGELOG.md#500-2020-05-12) | [**Migration from v4**](https://akveo.github.io/react-native-ui-kitten/docs/migration/4x500-migration#4x--500-ui-kitten-migration)

**Comment by malashkevich** [comment]
## 5.1.0 (2021-04-16)

### :rocket: Highlights:
* Added `onMonthChange` function to the Calendar component ([d70d978](https://github.com/akveo/react-native-ui-kitten/commit/d70d978))
* Added `ghost` appearance for ButtonGroup component (#1382) ([849fb48](https://github.com/akveo/react-native-ui-kitten/commit/849fb48)), closes [#1382](https://github.com/akveo/react-native-ui-kitten/issues/1382)
* Improved input component caption structure (#1349) ([862a12f](https://github.com/akveo/react-native-ui-kitten/commit/862a12f)), closes [#1349](https://github.com/akveo/react-native-ui-kitten/issues/1349)
* ModalService can adjust position of items if StatusBar on Android is translucent  (#1372) ([407356e](https://github.com/akveo/react-native-ui-kitten/commit/407356e)), closes [#1372](https://github.com/akveo/react-native-ui-kitten/issues/1372) [#743](https://github.com/akveo/react-native-ui-kitten/issues/743)
* ViewPager now has `swipeEnabled` prop ([c9e791f](https://github.com/akveo/react-native-ui-kitten/commit/c9e791f))

### :rocket: Features
* Update input styles when view & flex props are intersects each other (#1351) ([b9dbd58](https://github.com/akveo/react-native-ui-kitten/commit/b9dbd58)), closes [#1351](https://github.com/akveo/react-native-ui-kitten/issues/1351)

### :bug: Bug Fixes
* **Modal**: Add change orientation event with modal render (#1346) ([15fb285](https://github.com/akveo/react-native-ui-kitten/commit/15fb285)), closes [#1346](https://github.com/akveo/react-native-ui-kitten/issues/1346)
* **Datepicker**: Make `autodismiss` prop optional ([123eb2b](https://github.com/akveo/react-native-ui-kitten/commit/123eb2b))
* **Popover**: Popover now reacts to Android hardware back button press ([2a103ed](https://github.com/akveo/react-native-ui-kitten/commit/2a103ed))
* **ViewPager**: Now skips if page is null ([c49753b](https://github.com/akveo/react-native-ui-kitten/pull/1165/commits/c49753b22167e9d182db7cedc5c8825d0fe3b5b2))
* **Input**: Fixed cases when input does not detect press around the edges ([e992b66](https://github.com/akveo/react-native-ui-kitten/pull/1165/commits/e992b66fa27036bc85407a830c1cfca0708247a0))
* **Autocomplete**: Async Autocomplete does not show options initially ([2ca80e1](https://github.com/akveo/react-native-ui-kitten/pull/1165/commits/2ca80e1caffd30ddde9f2d183e34d67b01f9e514))
